minesweeper.py

```diff
@@ -0,0 +1,196 @@
+from collections import defaultdict
+from random import sample
+
+
+class Square(object):
+
+    def __init__(self):
+        """Creates minesweeper square with mine and clicked attributes."""
+        self.clicked = False
+        self.flagged = False
+        self._mine = False
+
+    def __repr__(self):
+        return "<Sq clicked: {}>".format(self.clicked, self._mine)
+
+
+class Board(object):
+
+    def __init__(self, mine_num=20, height=10, width=10):
+        """Creates a 10x10 board with specified number of mines."""
+        self.board = defaultdict(dict)
+        self.height = height
+        self.width = width
+        self.mine_num = mine_num
+        self.square_qty = height * width
+
+        mine_locs = sample(range(self.square_qty), mine_num)
+
+        for i in range(height):
+            for j in range(width):
+                self.board[i][j] = Square()
+
+        for mine in mine_locs:
+            row = mine / height
+            col = mine % width
+
+            self.board[row][col]._mine = True
+
+    def __repr__(self):
+        return "<{}>".format(self.board)
+
+    def _count_touching_mines(self, row, col):
+        """Returns the number of mines surrounding given square."""
+        neighbors = []
+        mines_qty = 0
+
+        for i in range(-1, 2):
+            row_index = row + i
+            if row_index >= 0 and row_index < self.height:
+                if col > 0:
+                    neighbors.append((row_index, col - 1))
+                if col < self.width - 1:
+                    neighbors.append((row_index, col + 1))
+                neighbors.append((row_index, col))
+
+        for neighbor in neighbors:
+            if self.board[neighbor[0]][neighbor[1]]._mine:
```

Comment:
Seems like this check could be in the above for loop. No need to store neighbors in an array.